### PR TITLE
plugins/main/ptp: add functional tests

### LINKED ISSUE #260: ptp: no functional tests
The `ptp` main plugin doesn't have any functional tests. This is fairly easy to do was the bridge or macvlan tests can be used as a template. 

@tomdee want to give this a try?

### PR DESCRIPTION
**Pull request by feiskyer**
Fixes #260

### CODE REVIEW

**Comment by feiskyer** [comment]
CC @steveeJ

File: plugins/main/ptp/ptp_test.go [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+		defer targetNs.Close()
+
+		args := &skel.CmdArgs{
+			ContainerID: "dummy",
+			Netns:       targetNs.Path(),
+			IfName:      IFNAME,
+			StdinData:   []byte(conf),
+		}
+
+		// Make sure ptp link exists in the target namespace
```
Comment (steveeJ) [inline plugins/main/ptp/ptp_test.go]: This comment is accurate as there's no check for the link in the target NS but rather the plugin call from the original NS. Maybe:

// Execute the plugin, creating the PTP endpoints

File: plugins/main/ptp/ptp_test.go [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+		err = targetNs.Do(func(ns.NetNS) error {
+			defer GinkgoRecover()
+
+			link, err := netlink.LinkByName(IFNAME)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(link.Attrs().Name).To(Equal(IFNAME))
+			return nil
+		})
+		Expect(err).NotTo(HaveOccurred())
+
```
Comment (steveeJ) [inline plugins/main/ptp/ptp_test.go]: // Call the plugins with the DEL command, deleting the veth endpoints

**Comment by feiskyer** [comment]
@steveeJ Updated. PTAL.

**Comment by steveeJ** [comment]
Thanks @feiskyer, this LGTM!

@tomdee, please do the 2nd review and merge eventually!

**Comment by tomdee** [comment]
Looks fine. Merging.

**Pull request merged**